src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2330,6 +2330,21 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                 break;
             }
         }
+        case SMART_ACTION_SET_CORPSE_DELAY:
+        {
+            ObjectList* targets = GetTargets(e, unit);
+            if (!targets)
+                break;
+
+            for (ObjectList::const_iterator itr = targets->begin(); itr != targets->end(); ++itr)
+            {
+                if (IsCreature(*itr))
```

Comment:
this kind of check usually stores the result of ToCreature() to a variable in the if ( ) check and then uses it, unless I'm missing something particular about IsCreature() compared to ToCreature()